cpu/nrf52/radio/nrf802154/nrf802154_radio.c

```diff
@@ -446,7 +459,7 @@ int nrf802154_init(void)
     timer_stop(NRF802154_TIMER);
 
     /* power off peripheral */
-    NRF_RADIO->POWER = 0;
+    _power_off();
```

Comment:
When this is called, `_hfxo_requests` is already 0, so it will underflow.
Either keep `NRF_RADIO->POWER = 0;` here or add

Reply:
Thats bad, it means that the driver in its current form can break the power management for third party modules... Example: 
- some other module explicitly requests the HFXO
- this drivers init is called -> the HFXO is disabled
- the other module will not function as expected as it depends on HFXO

Bottom line: we need to make sure that each driver/module requesting the HFXO is always keeping the HFXO in a consistent state. So in case there are user API functions that can alter the HFXO state and can be called any time (e.g. `nrf802154_ops->off`), the driver needs to keep its own state for the HFXO.